src/replica/Configuration.cc

```diff
@@ -735,20 +742,52 @@ DatabaseInfo& Configuration::_databaseInfo(util::Lock const& lock, string const&
 }
 
 DatabaseInfo& Configuration::_publishDatabase(util::Lock const& lock, string const& name, bool publish) {
-    DatabaseInfo& databseInfo = _databaseInfo(lock, name);
-    if (publish && databseInfo.isPublished) {
-        throw logic_error(_context(__func__) + " database '" + databseInfo.name + "' is already published.");
-    } else if (!publish && !databseInfo.isPublished) {
-        throw logic_error(_context(__func__) + " database '" + databseInfo.name + "' is not published.");
-    }
-    if (_connectionPtr != nullptr) {
-        _connectionPtr->executeInOwnTransaction([&](decltype(_connectionPtr) conn) {
-            conn->executeSimpleUpdateQuery("config_database", conn->sqlEqual("database", databseInfo.name),
-                                           make_pair("is_published", publish ? 1 : 0));
-        });
+    DatabaseInfo& databaseInfo = _databaseInfo(lock, name);
+    if (publish && databaseInfo.isPublished) {
+        throw logic_error(_context(__func__) + " database '" + databaseInfo.name + "' is already published.");
+    } else if (!publish && !databaseInfo.isPublished) {
+        throw logic_error(_context(__func__) + " database '" + databaseInfo.name + "' is not published.");
+    }
+    if (publish) {
+        uint64_t const publishTime = PerformanceUtils::now();
+        // Firstly, publish all tables that were not published.
+        for (auto const& table : databaseInfo.tables()) {
+            if (_connectionPtr != nullptr) {
+                _connectionPtr->executeInOwnTransaction([&](decltype(_connectionPtr) conn) {
```

Comment:
default capture modes of lambda expressions make me nervous.
https://learning.oreilly.com/library/view/effective-modern-c/9781491908419/ch06.html#:-:text=Item%2031%3A%E2%80%82Avoid%20default,may%20not%20be).
(you can log in with your Stanford credentials)
YMMV.